### DISCUSSION TITLE
Can we add custom testnetchain/network to rainbow kit wallet via frontend button which says "add to rainbowkit"

### DISCUSSION THREAD

**Discussion by BornPsych** [rainbow-me/rainbowkit#1845] [Q&A] [answered]
Can we add custom testnetchain/network to rainbow kit wallet via frontend button which says "add to rainbowkit"

**Comment by 0xmkh** [comment, marked as answer]
@BornPsych You can check if wallet is connected then switch to the network by using `useSwitchChain` hook from wagmi here: https://wagmi.sh/react/api/hooks/useSwitchChain#useswitchchain

This will either add the network if doesn't exist or switch to it. I think instead of having "Add to rainbowkit" you would rather have "Add {chain}", or "Switch to {chain}".
